types/ui/activeEffectConfig.d.ts

```diff
@@ -1,4 +1,50 @@
-declare class ActiveEffectConfig extends FormApplication {
-  // TODO Declare class
-  _updateObject(event?: Event, formData?: object): never;
+/**
+ * A form designed for creating and editing an Active Effect on an Actor or Item entity.
+ *
+ * @param object  - The target active effect being configured
+ * @param options - Additional options which modify this application instance
+ * @typeParam D - The data structure used to render the handlebars template.
+ * @typeParam O - the type of the ActiveEffect which should be managed by this form sheet
+ */
+declare class ActiveEffectConfig<
+  D extends object = ActiveEffectConfig.Data,
+  O extends ActiveEffect = D extends ActiveEffectConfig.Data<infer T> ? T : ActiveEffect
+> extends FormApplication<D, O> {
+  /** @override */
+  static get defaultOptions(): FormApplication.Options;
+
+  /** @override */
+  get title(): string;
+
+  /** @override */
+  getData(options?: Application.RenderOptions): D | Promise<D>;
+
+  /** @override */
+  activateListeners(html: JQuery): void;
+
+  /**
+   * Provide centralized handling of mouse clicks on control buttons.
+   * Delegate responsibility out to action-specific handlers depending on the button action.
+   * @param event - The originating click event
+   */
+  protected _onEffectControl(event: JQuery.ClickEvent): this | void;
```

Comment:
Is this really that specific, that it has to be a JQuery event?

Reply:
Well, in theory, the method can probably handle more (possibly any event).However, the JSDoc actually says this would be a `MouseEvent` but it really should only be click events (the JSDoc says "The originating click event"). And it's only ever called with `JQuery.ClickEvent`s:
```ts
  activateListeners(html) {
    super.activateListeners(html);
    html.find(".effect-control").click(this._onEffectControl.bind(this));
  }
 ```
 (`html` is always of type `JQuery`, as `HTMLElement` doesn't even have `find`)

So I think `JQuery.ClickEvent is actually the right thing.